### PR TITLE
Add test cases for HexStr (std::reverse_iterator and corner cases)

### PR DESCRIPTION
**Pull request by kostaz**

### CODE REVIEW

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/12570/commits/ac48861815832e924bc355553f677f54ace984c5, thanks for adding tests!

File: src/test/util_tests.cpp [hunk 1]
```diff
@@ -94,6 +94,20 @@ BOOST_AUTO_TEST_CASE(util_HexStr)
...
+               ParseHex_expected + sizeof(ParseHex_expected)),
+        "");
+
+    BOOST_CHECK_EQUAL(
+        HexStr(ParseHex_expected + sizeof(ParseHex_expected),
+               ParseHex_expected + sizeof(ParseHex_expected), true),
+        "");
+
+    BOOST_CHECK_EQUAL(
+        HexStr(ParseHex_expected, ParseHex_expected),
```
Comment (promag) [inline src/test/util_tests.cpp]: This is already tested above (L97-100), where `begin == end`, or am I missing something? (I prefer this over the above).

File: src/test/util_tests.cpp [hunk 2]
```diff
@@ -117,6 +117,58 @@ BOOST_AUTO_TEST_CASE(util_HexStr)
...
+        "b0fd8a6704"
+    );
+
+    BOOST_CHECK_EQUAL(
+        HexStr(std::reverse_iterator<const uint8_t *>(ParseHex_expected + 5),
+               std::reverse_iterator<const uint8_t *>(ParseHex_expected), true),
+        "b0 fd 8a 67 04"
+    );
+
+    BOOST_CHECK_EQUAL(
```
Comment (promag) [inline src/test/util_tests.cpp]: This is already tested in L155-L159.

File: src/test/util_tests.cpp [hunk 3]
```diff
@@ -117,6 +117,58 @@ BOOST_AUTO_TEST_CASE(util_HexStr)
...
+        "04"
+    );
+
+    BOOST_CHECK_EQUAL(
+        HexStr(std::reverse_iterator<const uint8_t *>(ParseHex_expected + 5),
+               std::reverse_iterator<const uint8_t *>(ParseHex_expected)),
+        "b0fd8a6704"
+    );
+
+    BOOST_CHECK_EQUAL(
```
Comment (promag) [inline src/test/util_tests.cpp]: This is already tested in L143-L146.

**Review by promag — COMMENTED** [review]
There are some duplicate tests, which only the "length" changes.

Post utACK.

**Pull request merged**